File: C++/프로그래머스/0/120818. 옷가게 할인 받기/옷가게 할인 받기.cpp
using namespace std;

int solution(int price) {
    int p = price / 100000;
    if (p > 4) return (int)(price * 0.8);
    if (p > 2) return (int)(price * 0.9);
    if (p > 0) return (int)(price * 0.95);
    return price;
}

File: Go/프로그래머스/0/120818. 옷가게 할인 받기/옷가게 할인 받기.go
func solution(price int) int {
    var temp_price float32
    
    switch v := price / 100000; {
        case v > 4:
            temp_price = float32(price) * 0.8
        case v > 2:
            temp_price = float32(price) * 0.9
        case v > 0:
            temp_price = float32(price) * 0.95
        default:
            temp_price = float32(price)
    }
    
    return int(temp_price)
}